packages/instantsearch.js/src/lib/routers/history.ts

```diff
@@ -207,11 +207,27 @@ class BrowserHistory<TRouteState> implements Router<TRouteState> {
    * See: https://github.com/algolia/instantsearch.js/issues/790
    */
   public createURL(routeState: TRouteState): string {
-    return this._createURL({
+    const url = this._createURL({
```

Comment:
I can see the risk (and little perf impact), but personally I'd also be happy with `new URL(this._createURL(), location.href)` (and making sure everything `URL` as well as `string`

Reply:
I tried with this : 
```javascript
public createURL(routeState: TRouteState): string {
    const location = this.getLocation();
    return new URL(
      this._createURL({
        qsModule: qs,
        routeState,
        location,
      }),
      location.href
    ).href;
  }
}
```
And it seems to work well, even with SSR provided their `getLocation` does return a URL.

However I just realized that `URL` surprisingly [does not work in IE11](https://caniuse.com/?search=URL) 🤦

Reply:
We're lucky we're only using it in tests then! I indeed forgot URL is not compatible with IE. In a next major we could do that to simplify things then (use and accept URL everywhere, cast to URL in createURL)